packages/new-editor/src/utils/tree.ts

```diff
@@ -0,0 +1,73 @@
+import type { Row, Table } from '@tanstack/react-table';
+import { addNode, getNode, removeNode } from '../data/data';
+import type { TreeNode, TreePath, addChildToFirstSelectedRowReturnType } from '../types/config';
+import { getFirstSelectedRow, selectRow } from './table';
+
+export const addChildToFirstSelectedRow = <TNode extends TreeNode<TNode>>(
```

Comment:
Should these become part of our UI widgets?

Reply:
Maybe. First time of such a method. 👍 
The inscription view only knows adding in a data table not trees.